src/scenes/Organizations/Detail/EditOrganization/EditOrganization.tsx

```diff
@@ -0,0 +1,60 @@
+import { useSnackbar } from 'notistack';
+import React, { FC } from 'react';
+import { Scalars, UpdateOrganizationInput } from '../../../../api';
+import { DialogState } from '../../../../components/Dialog';
+import { DialogForm } from '../../../../components/Dialog/DialogForm';
+import { SubmitError, TextField } from '../../../../components/form';
+import { OrganizationDocument } from '../OrganizationDetail.generated';
+import { useUpdateOrganizationMutation } from './EditOrganization.generated';
+
+type EditOrganizationFormProps = DialogState & { orgId: Scalars['String'] };
+
+export const EditOrganizationForm: FC<EditOrganizationFormProps> = (props) => {
+  const { orgId, ...rest } = props;
+  const [updateOrg] = useUpdateOrganizationMutation();
+  const { enqueueSnackbar } = useSnackbar();
+  const submit = async (input: UpdateOrganizationInput) => {
+    const organizationInputWithId = {
+      organization: {
+        ...input.organization,
+        id: orgId,
+      },
+    };
+    const res = await updateOrg({
+      variables: { input: organizationInputWithId },
+      refetchQueries: [
+        {
+          query: OrganizationDocument,
+          variables: { input: orgId },
+        },
+      ],
```

Comment:
@richardgirges Do we need to refetch the query if the updated state is returned in the mutation result? Can't we save that round trip?

Reply:
We don't need to refetch the query in this case. All we have to do is ensure that the mutation response:
1. is returning the `id` property
2. is returning the same fragment as the query

apollo will automatically see that this mutated record has been cached previously and will auto-update the cache for that record.

Reply:
good to know, removed